packages/edit-site/src/components/header-edit-mode/index.js

```diff
@@ -77,7 +79,12 @@ export default function HeaderEditMode() {
 	}, [] );
 
 	const isLargeViewport = useViewportMatch( 'medium' );
-	const isTopToolbar = ! isZoomOutMode && hasFixedToolbar && isLargeViewport;
+	const { showFixedToolbar } = useShowBlockTools();
+	const hasTopToolbar =
```

Comment:
same comment as above, could this determination be abstracted in tiny hook?

Reply:
I refactored it further by adding `! isZoomOutMode` to the `showFixedToolbar` check and removing `hasFixedToolbar` here, as that setting was already checked in `showFixedToolbar`. Now the `hasTopToolbar` check is just `isLargeViewport && showFixedToolbar`. 

I don't think it's worth extracting that to its own hook as we can't easily share it between the `edit-site` and `edit-post` packages. Especially now that it's just an `isLargeViewport` check. I didn't want to put the `showTopToolbar` vs `showFixedToolbar` within the `block-editor` package, as it isn't supposed to have knowledge about how the block toolbar is being used outside of it.